### PR TITLE
Settings: use fsync() for more robust setting files saving

### PR DESCRIPTION
**Pull request by poire-z**
Bump base for util.fsyncOpenedFile() and util.fsyncDirectory() https://github.com/koreader/koreader-base/pull/1021

Use these to force flush to storage device when saving luasetting, docsetting, and history.lua.
Also dont rotate them to .old until they are at least 60 seconds old.
Discussion in #3625.

(Unrelated, but I also updated Luadata so it does not output stuff when there is no file - I used to see a few lines as I got not custom multiswiple.lua defined.)

Also bump crengine: open (as text) small or empty files https://github.com/koreader/crengine/pull/322

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/5678)

### CODE REVIEW

File: frontend/apps/reader/modules/readerfooter.lua [hunk 1]
```diff
@@ -43,12 +43,19 @@ local symbol_prefix = {
     letters = {
         time = nil,
         pages_left = "=>",
+        -- @translators This is the footer letter prefix for battery % remaining.
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readerfooter.lua]: Looks good :)

**Comment by poire-z** [comment]
@robert00s : can you confirm a nightly with this PR does not crash on Kindle?
I tested on Kobo and Android, but I can't test Kindles. @NiLuJe : do all Kindle on any firmware use that fuse layer you mentionned a few times?

**Comment by NiLuJe** [comment]
Yup, all of 'em ;/.

**Comment by robert00s** [comment]
@poire-z 
I tested yesterday on Kindle. No problem so far :)

**Pull request merged**